gfsphysics/physics/aerinterp.f90

```diff
@@ -1,165 +1,187 @@
-       call nf_get_var(ncid, varid, delp_clm)
-
-!      if ( imon == 1 .and.  me == master ) then
-!        print *, 'EJ, ps_clm:', ps_clm(1,1,1)
-!        print *, 'EJ, delp_clm:', delp_clm(1,1,:,1)
-!      endif
+       call nf_get_var(ncid, varid, buff) 
 
-       do i = 1, lonsaer
        do j = 1, latsaer
+        do i = 1, lonsaer
+! constract pres_tmp (top-down), note input is top-down
+         pres_tmp(i,1) = 0.
+         do k=2, dim3
+          pres_tmp(i,k) = pres_tmp(i,k-1)+buff(i,j,k)
+         enddo    !k-loop 
+        enddo     !i-loop (lon)
 
-! constract pres_tmp (top-down)
-        pres_tmp(1) = 0.
-        do k=2, lmerra
-         pres_tmp(k) = pres_tmp(k-1) + delp_clm(i,j,k,1)
-        enddo
-!       if (imon==1 .and.  me==master .and. i==1 .and. j==1 ) then
-!        print *, 'EJ, pres_tmp:', pres_tmp(:)
-!       endif
-
-! extract pres_tmp to fill aer_pres
+! extract pres_tmp to fill aer_pres (in  Pa)
         do k = 1, levsaer
          if ( iflip == 0 )  then             ! data from toa to sfc
            klev = k
          else                                ! data from sfc to top
-           klev = ( lmerra - k ) + 1
+           klev = ( dim3 - k ) + 1
          endif 
-         aer_pres(i,j,k,imon)=  pres_tmp(klev)
+         do i = 1, hmx
+         aer_pres(i+hmx,j,k,imon)= 1.d0*pres_tmp(i,klev)
+         aer_pres(i,j,k,imon)    = 1.d0*pres_tmp(i+hmx,klev)
+         enddo     !i-loop (lon)
         enddo     !k-loop (lev)
-!       if (imon==1 .and.  me==master .and. i==1 .and. j==1 ) then
-!        print *, 'EJ, aer_pres:', aer_pres(i,j,:,imon) 
-!       endif
-
        enddo     !j-loop (lat)
-       enddo     !i-loop (lon)
 
-!      if (imon==1 .and.  me==master ) then
-!        print *, 'EJx, aer_pres_i1:',(aer_pres(1,1:180,levsaer,imon) )
-!      endif
+! ====> construct 4-d aerosol array (kg/kg)
+! merra2 data is top down
+! for GFS, iflip 0: toa to sfc; 1: sfc to toa
+       DO ii = 1, ntrcaerm
+         vname=trim(specname(ii))
+         call nf_inq_varid(ncid, vname, varid)
+         call nf_get_var(ncid, varid, buffx)
 
-! construct lat/lon array
-       if (imon == 1 ) then
-        call nf_inq_varid(ncid, "lat", varid)
-        call nf_get_var(ncid, varid, aer_lat)
-        call nf_inq_varid(ncid, "lon", varid)
-        call nf_get_var(ncid, varid, aer_lon)
-        do i = 1, lonsaer
-          if(aer_lon(i) < 0.)  aer_lon(i) = aer_lon(i) + 360.
-        enddo
-!       if (imon==1 .and. me == master) then
-! print *, "EJ, lat:", aer_lat(:)
-! print *, "EJ, lon:", aer_lon(:)
-!       endif
-       endif
+         do j = 1, latsaer
+         do k = 1, levsaer 
+! input is from toa to sfc
+          if ( iflip == 0 )  then             ! data from toa to sfc
+            klev = k
+          else                                ! data from sfc to top
+            klev = ( dim3 - k ) + 1
+          endif 
+          do i = 1, hmx
+          aerin(i+hmx,j,k,ii,imon) = 1.d0*buffx(i,j,klev,1)
```

Comment:
do we really need this multiplication by 1.0d0?  There are several places like this.